src/core/manager.c

```diff
@@ -2638,21 +2642,24 @@ static bool manager_process_barrier_fd(char * const *tags, FDSet *fds) {
 static void manager_invoke_notify_message(
                 Manager *m,
                 Unit *u,
+                PidRef *pidref,
                 const struct ucred *ucred,
                 char * const *tags,
                 FDSet *fds) {
 
         assert(m);
         assert(u);
+        assert(pidref_is_set(pidref));
         assert(ucred);
+        assert(pidref->pid == ucred->pid);
```

Comment:
this is data coming from outside, it should probably be a check and return rather than an asser

Reply:
`manager_dispatch_notify_fd` would refuse that case already?